bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/SurroundWithBracketsStrategy.java

```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+ * Copyright (c) ETAS GmbH 2024, all rights reserved.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     ETAS GmbH - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jface.text;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.swt.SWT;
+
+import org.eclipse.jface.text.source.ISourceViewer;
+
+/**
+ * @since 3.26 This strategy supports surrounding the selected text with similar opening and closing
+ *        brackets when the text is selected and an opening bracket is inserted.
+ */
+public class SurroundWithBracketsStrategy implements IAutoEditStrategy {
+
+	private ISourceViewer sourceViewer;
+
+	@SuppressWarnings("nls")
+	private final static Map<String, String> bracketsMap= new HashMap<>(Map.of("(", ")", "[", "]", "{", "}", "<", ">", "\"", "\"", "'", "'", "`", "`"));
```

Comment:
Must it really be a modifiable (static) map here?!?

Reply:
+1
Using a `Map.of()` construct here would probably fit better.

Reply:
Can it be changed to 

`private final Map<String, String> bracketsMap= Collections.unmodifiableMap(Map.of("(", ")", "[", "]", "{", "}", "<", ">", "\"", "\"", "'", "'", "`", "`"));`

Reply:
Map.of's result is already unmodifiable.